### ISSUE TITLE
Getting type related when running Stryker

### ISSUE THREAD

**Issue by barrykaplan** [szhsin/react-menu#1100]
We use Stryker for mutation testing. Today we started to use react-menu. Now we are getting errors form Stryker of the form
```
18:40:27 (1252513) ERROR Stryker Unexpected error occurred while running Stryker StrykerError: Error: An error occurred during initialization of the "typescript" checker. Inner error: Error: Typescript error(s) found in dry run compilation: src/styles/components/StyleControlledMenu.tsx(6,7): error TS4023: Exported variable 'CellMenu' has or is using name 'BaseProps' from external module "/home/bkaplan/dev.hbk/reliasoft/reli.ui/node_modules/@szhsin/react-menu/types/index" but cannot be named.
```

Any clues as to what is going would be helfpul.

This is from usage:
```
import styled from 'styled-components'
import { MainTheme } from '../Theme'
import { ControlledMenu } from '@szhsin/react-menu'
import '@szhsin/react-menu/dist/index.css'

const CellMenu = styled(ControlledMenu)`
  ul {
    margin: 0;
    padding: 0;
  }

  li {
    padding: 0.2em 0.5em;
    font-size: 15px;
    background-color: #fff;
    color: ${MainTheme.colors.colorMid};

    &:hover {
      background-color: ${MainTheme.colors.colorLight};
      color: ${MainTheme.colors.colorBlack};
    }

    &:active {
      background-color: ${MainTheme.colors.colorLight};
    }

    &:focus-visible {
      background-color: ${MainTheme.colors.colorTransparent};
    }

    .MuiTouchRipple-rippleVisible {
      color: ${MainTheme.colors.colorLight} !important;
    }
  }

  .szh-menu__divider {
    background-color: ${MainTheme.colors.colorTransparent};
    padding: 0em;
    border-bottom: 1px solid ${MainTheme.colors.colorLight};
    margin: 0.2em 0;
  }

  .szh-menu__item--disabled {
    color: ${MainTheme.colors.colorDarkGray};
    background-color: ${MainTheme.colors.colorLightGray};

    &:hover {
      background-color: ${MainTheme.colors.colorLightGray};
      color: ${MainTheme.colors.colorDarkGray};
    }
`

export default CellMenu
```

**Comment by szhsin** [comment]
Not sure what's causing this error but there were discussions around similar issues on [stackoverflow](https://stackoverflow.com/questions/43900035/ts4023-exported-variable-x-has-or-is-using-name-y-from-external-module-but)  and [github](https://github.com/microsoft/TypeScript/issues/5711).

One reason could be the `BaseProps` interface is not exported and not accessible. It may cause the error in some Typescript configurations.

What we can try is to export `BaseProps` from this package and check if the error goes away. You can try it by installing this alpha release:
``` bash
npm i @szhsin/react-menu@4.0.4-alpha.0
```
Let us know if this fixes the issue, or if not, does it produce a different error?

**Comment by szhsin** [comment]
That's great! thanks for confirming. Please give a try for another pre-release which has all the types exported:
```bash
npm i @szhsin/react-menu@4.0.4-alpha.1
```

**Comment by barrykaplan** [comment]
That did it, thanks!

**Comment by szhsin** [comment]
Fixed by [v4.1.0](https://github.com/szhsin/react-menu/releases/tag/v4.1.0)
